sla_per_use_system_test.py

```diff
@@ -0,0 +1,394 @@
+        if scylla_load < waited_cpu_min:
+            self.log.warning('Load isn\'t high enough. The test results may be not correct')
+
+        return results
+
+    def clean_auth(self, entities_list_of_dict):
+        for entity in entities_list_of_dict:
+            sl = entity.get('service_level')
+            role = entity.get('role')
+            user = entity.get('user')
+            if user:
+                user.drop()
+            if role:
+                role.drop()
+            if sl:
+                sl.drop()
+
+        self.backgroud_task = None
+
+    # TODO: decide if need that
+    # def create_background_load(self, duration='120m', timeout=900, cpu_minimum=99, sleep_time_sec=120,
+    #                            rows_increase=500000):
+    #     stress_cmd = self.STRESS_WRITE_DURATION_CMD.format(duration=duration, user=self.background_user[0],
+    #                                                        password=self.background_user[0],
+    #                                                        threads=250, pop='%d..%d' % (self.n, self.n+rows_increase))
+    #     self.backgroud_task = self.run_stress_thread(stress_cmd=stress_cmd)
+    #
+    #     return wait.wait_for(func=self.wait_for_load_reached, step=sleep_time_sec, timeout=timeout,
+    #                          cpu_minimum=cpu_minimum, sleep_time_sec=sleep_time_sec)
+    #
+    # def wait_for_load_reached(self, cpu_minimum=99, sleep_time_sec=120):
+    #     end = int(time.time())
+    #     start = int(time.time())-sleep_time_sec
+    #     cpu = self.prometheus_stats.get_scylla_reactor_utilization(start_time=start, end_time=end)
+    #     if cpu and cpu > cpu_minimum:
+    #         self.log.debug("CPU load reached {}%".format(cpu_minimum))
+    #         return cpu > cpu_minimum
+    #     return False
+
+    def test_run_all(self):
+        """
+        Start all tests one by one
+        """
+        session = self.prepare_schema()
+
+        # Sleep after writes and before read load
+        time.sleep(300)
+        results = {'sla_per_user_basic': self.sla_per_user_basic(session=session)}
+
+        # Sleep between tests
+        time.sleep(300)
+        results['throughput_latency_load'] = self.throughput_latency_load(session=session)
+
+        failed_tests = ['Test {}. Error: {}'.format(test, result) for test, result in results.iteritems() if result]
+        self.assertFalse(failed_tests,
+                         msg='Failed tests: {}'.format('\n'.join(test for test in failed_tests)))
+
+    def sla_per_user_basic(self, session):
+        """
+        Basic test
+        - Add SLA and grant to user (before any load)
+        - user200 with 200 shares
+        - user1000 qith 1000 shares
+        - Each user runs load from own loader (round robin)
+        - Expect OPS ratio between two loads is 1:5 (e.g. 200:1000)
+        - Expect scheduler run time between two loads is 1:5 (e.g. 200:1000)
+        """
```

Comment:
If we want to replicate this test with different shares per user, shall we replicate the entire code?

Reply:
Same If we want to test with 3 or more users.

Reply:
Same if we want to run from cache_only, disk_only and mix_of_them

Reply:
The tests are built using helpers. So it should be easy to duplicate them. 
I can do it more universal, but it will be less readable

Reply:
> If we want to replicate this test with different shares per user, shall we replicate the entire code?

I did test body as hepler for case to run same test with different shares and load type(cache onle, disk only, mixed

Reply:
> Same If we want to test with 3 or more users.

For this case I think it's better to add another helper